src/controllers/projectCtrl.js

```diff
@@ -37,66 +37,100 @@ exports.deleteProject = (req, res) => {
 };
 
 exports.updateProjectInfos = (req, res) => {
-  Project.findByIdAndUpdate(req.params.projectId, req.body, {
-    new: true,
-  })
+  //check if project exist
+  Project.findById(req.params.projectId)
     .then((project) => {
-      if (!project)
-        res
-          .status(400)
-          .json({ message: 'No Item updated: The project does not exist' });
-      else
-        res.status(200).json({
-          message: 'The project has been modified correctly!',
-          data: project,
+      if (project) {
+        project = {
+          ...project,
+          ...req.body,
+        };
+        project
+          .save()
+          .then((newProject) =>
+            res.status(200).json({
+              message: 'Project infos updated!',
+              newProject,
+            })
+          )
+          .catch((error) => res.status(500).json(error));
+      } else {
+        res.status(404).json({
+          message: 'Project not found!',
         });
+      }
     })
     .catch((error) =>
       res.status(401).json({ message: 'Invalid Request!', error })
     );
 };
 
 exports.addUserToProject = (req, res) => {
-  Project.findByIdAndUpdate(
-    req.params.projectId,
-    {
-      $push: {
-        users: {
-          id: req.body.userId,
-          role: req.body.role,
-        },
-      },
-    },
-    {
-      new: true,
-      upsert: true,
-    }
-  )
+  //check if project exist
+  Project.findById(req.params.projectId)
```

Comment:
Why did you replace the findByIdAndUpdate by findById and update within its function manually?

Reply:
To add a test to check if user not already in project

Reply:
@remirubis